### ISSUE TITLE
offset causes model gof error

### ISSUE THREAD

**Issue by netterie** [statnet/ergm.ego#14]
Is there something about this particular model that is causing the offset to malfunction?

```r
data(faux.mesa.high)
fmh.ego <- as.egodata(faux.mesa.high)
model = ergm.ego(fmh.ego ~ edges + offset(nodematch("Sex", diff=FALSE)), 
                offset.coef=c(-Inf),
                popsize=network.size(faux.mesa.high))
    
gof(model, GOF='model') # gives same error as 
summary(fmh.ego ~ edges + offset(nodematch("Sex", diff=FALSE)))
```

Error is: _Error in eval(expr, envir, enclos) : could not find function "EgoStat.offset"_

**Comment by dth2** [comment]
I just ran through your example and the offset is not malfunctioning.  If you just run a summary on the model it works.  The issue is that ergm.ego doesn’t recognize offset as a model term so the summary and gof functions specific to ergm.ego are throwing an error.  EgoStat is the function in ergm.ego that goes through the ergm equation and decided what to do with each term in the model.  The only terms that ergm.ego will recognize are listed below.  I think this is an issue for Pavel.  Hopefully it is a quick fix.  

 


EgoStat.edges

	
#' EgoStat.nodecov EgoStat.nodefactor EgoStat.nodematch EgoStat.nodemix

	
#' EgoStat.absdiff EgoStat.degree EgoStat.degrange EgoStat.concurrent

	
#' EgoStat.concurrentties EgoStat.degreepopularity EgoStat.mean.age

 

 

https://github.com/statnet/ergm.ego-private/blob/master/R/EgoStat.R

 

 

> summary(model)

 

==========================

Summary of model fit

==========================

 

Formula:   fmh.ego ~ edges + offset(nodematch("Sex", diff = FALSE))

 

Iterations:  5 out of 20 

 

Monte Carlo MLE Results:

              Estimate Std. Error MCMC % p-value    

netsize.adj    0.00000    0.00000      0      NA    

edges         -3.92546    0.07474      0  <1e-04 ***

nodematch.Sex     -Inf    0.00000      0  <1e-04 ***

---

Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

 

 

The following terms are fixed by offset and are not estimated:

  netsize.adj nodematch.Sex

**Comment by dth2** [comment]
Pavel,

It looks like you have addressed this in both the Public and Private versions which is awesome.  Is there an easy way to pull this change in to the version that SHAMP currently uses?

 

https://github.com/statnet/ergm.ego-private/tree/3.7-compat

**Comment by krivit** [comment]
I merged into `3.7-compat` as well.

**Comment by netterie** [comment]
Thank you Pavel!

**Comment by martinamorris** [comment]
> I merged into 3.7-compat as well.

thx for the quick response ;)
